### ISSUE TITLE
Stop using verbatim `"` and `;`

### ISSUE THREAD

**Issue by slotThe** [kmonad/kmonad#258]
Here's a radical thought: what if we stopped allowing users to use unescaped `"` and `;` in their configurations?  This would obviously break all existing configs, but perhaps it is worth it (with the right error message, of course).  It would have the advantage of making it _much_ easier for people who don't use the [GNU/Emacs](https://github.com/kmonad/kbd-mode) or [Vim](https://github.com/kmonad/kmonad-vim) modes to properly edit their config files.  They could then basically use any LISP mode if they don't use block comments, or their editors mode for `racket` if they do. 

Full disclosure: it would also free `kbd-mode` of some horrible hacks and possibly get it to a state where we could submit it to MELPA; though that shouldn't be the driving factor here.

**Comment by vherrmann** [comment]
Instead of a thought, i have a solution to this. We could use the r7rs quoting mechanism or the one of guile (https://www.gnu.org/software/guile/manual/html_node/Symbol-Read-Syntax.html). In think that has advantages in comparison to prefixing with a backslash, since imho one shouldn't be able to put special characters in „normal“ symbols. E.g.

`\; this should be a comment`

`(a \)`

i think ↑ should just be the list (a \ ).

**Comment by slotThe** [comment]
I think there's an argument to be made for still trying really hard to do "the expected thing".  We do have a configuration file and _not_ a proper LISP (see e.g. #237 for a discussion), so making some concessions seems quite sane to me.  Something like `\;` will (imo) always be preferrable to `col`, `sem`, `sc`, or whatever other crazy abbreviation we could come up with

**Comment by david-janssen** [comment]
> Here's a radical thought: what if we stopped allowing users to use unescaped " and ; in their configurations? 

Yeah, I was thinking the same actually. It would simplify a lot of things, and if we ever do want to include ubiquitous string literals this is basically going to have to be a must anyways.

However, I do not like making *any* back-support breaking change that doesn't go unaccompanied by a major version change. I'm currently doing a very extensive cleanup of the code, and I think we want to tinker with the parsers and command-line too. So, I suggest we make this change ready, but we don't merge it just yet. In a couple of days I'll probably move my `tinker` branch to a `keycode-refactor` on the `kmonad` org, (once Linux keyIO is working), then we cal cooperate to get Mac and Win support up and running too.

After that we should do a serious rethink of the documentation. At that point, I think a major version change is very warranted. So maybe start thinking about other back-support breaking changes you want to make, because after the major version change it's going to be no-back-support-breakage for a while again.

**Comment by david-janssen** [comment]
Oh, and in regards to the quoting mechanisms from scheme, I actually really like them, but I don't think they really solve the issue of how to insert references to keycodes inside of maps. Very cool for a programming language, though, but to #{"}# inside a map seems.. a bit too much :)
